tensorbay/cli/log.py

```diff
@@ -406,23 +510,41 @@ def _build_commit_tree(
                     break
 
                 # Save leaf node to leaf set.
-                if index == 0:
+                if not child_node:
                     leaves[commit_id] = current_node
                 # Save commit to commit dict.
                 commit_to_node[commit_id] = current_node
                 child_node = current_node
+            if show_drafts:
+                for draft in dataset_client.list_drafts(branch_name=revision):
+                    draft_node = _DraftNode(draft)
+                    leaves[draft_node.key] = draft_node
+                    self._key_to_branches[draft_node.key] = [draft.branch_name]
+                    parent_commit_id = draft.parent_commit_id
+                    if parent_commit_id == "":
+                        root_node = _RootCommitNode()
+                        if child_node:
+                            root_node.add_child(child_node)
+                        else:
+                            leaves[parent_commit_id] = root_node
+                        commit_to_node[parent_commit_id] = root_node
+                    commit_to_node[draft.parent_commit_id].add_child(draft_node)
+
+        return self._check_and_sort_leaves(leaves)
 
+    @staticmethod
+    def _check_and_sort_leaves(leaves: Dict[str, _T]) -> List[_T]:
         # Check the correction of leaf set.
-        for commit_id in tuple(leaves):
-            if leaves[commit_id].available_child_num != 0:
-                del leaves[commit_id]
-        return sorted(leaves.values(), key=lambda x: x.commit.committer.date, reverse=True)
+        for key, leaf_node in tuple(leaves.items()):
```

Comment:
Why use tuple here？

Reply:
The `dict` cannot be modified during iterating it